File: ios/ReactNativeModules/RandomModule.m
//
//  RandomModule.m
//  ReactNativeHybrid
//
//  Created by Oscar Cheung on 20/02/2023.
//

#import "React/RCTBridgeModule.h"

@interface RCT_EXTERN_MODULE(RandomModule, NSObject)

  RCT_EXTERN__BLOCKING_SYNCHRONOUS_METHOD(randSync)
  RCT_EXTERN_METHOD(rand: (RCTPromiseResolveBlock) resolve
                    rejecter: (RCTPromiseRejectBlock) reject)

@end

File: ios/ReactNativeModules/RandomModule.swift
//
//  RandomModule.swift
//  ReactNativeHybrid
//
//  Created by Oscar Cheung on 20/02/2023.
//

@objc(RandomModule)
class RandomModule: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool { return false }

  @objc
  func randSync() -> [Float] {
    return [Float.random(in: 0...1)]
  }

  @objc
  func rand(_ resolve: RCTPromiseResolveBlock,
            rejecter reject: RCTPromiseRejectBlock) -> Void {
    do {
      try resolve(Float.random(in: 0...1))
    } catch {
      reject("0", "Error occurred", error)
    }
  }

}
